utils/wasm-gen/src/generator/syscalls.rs

```diff
@@ -42,8 +42,19 @@ pub use additional_data::*;
 pub use imports::*;
 pub use invocator::*;
 
+use crate::generator::CallIndexesHandle;
 use gear_wasm_instrument::syscalls::{ParamType, SysCallName, SysCallSignature};
 
+/// Additional information about [`InvocableSysCall`].
+///
+/// Mainly used as an argument when building a sys-call.
+/// It is also possible to modify the signature while building sys-call.
+pub struct InvocableSysCallData {
```

Comment:
honestly, I don't this such "dto"s are suitable. It has only one purpose - carry data. The type has only three fields. Using that makes API more ambiguous. I'd get rid of it.